src/basic/audit-util.c

```diff
@@ -15,59 +15,96 @@
 #include "parse-util.h"
 #include "process-util.h"
 #include "socket-util.h"
+#include "stat-util.h"
 #include "user-util.h"
+#include "virt.h"
 
-int audit_session_from_pid(pid_t pid, uint32_t *id) {
-        _cleanup_free_ char *s = NULL;
-        const char *p;
-        uint32_t u;
+int audit_session_from_pid(const PidRef *pid, uint32_t *ret_id) {
```

Comment:
The first half of audit_session_from_pid() and audit_loginuid_from_pid() are equivalent except for the filename.
When you modernize the code, maybe better to split out that part?